src/annotator/annotation-sync.js

```diff
@@ -1,176 +1,116 @@
 
-      for (i = 0; i < bodies.length; i++) {
-        parsedAnnotations.push(this._parse(bodies[i]));
+    // Relay events from annotator to sidebar.
+    this._on('beforeAnnotationCreated', annotation => {
+      if (annotation.$tag) {
+        return;
       }
-      return parsedAnnotations;
-    }.call(this);
-    this._emit('annotationsLoaded', annotations);
-    return cb(null, annotations);
-  },
-};
+      this.bridge.call('beforeCreateAnnotation', this._format(annotation));
+    });
+  }
 
-// Handlers for events coming from this frame, to send them across the channel
-AnnotationSync.prototype._eventListeners = {
-  beforeAnnotationCreated: function (annotation) {
-    if (annotation.$tag) {
-      return undefined;
-    }
-    return this._mkCallRemotelyAndParseResults('beforeCreateAnnotation')(
-      annotation
+  /**
+   * Relay updated annotations from the annotator to the sidebar.
+   *
+   * This is called for example after annotations are anchored to notify the
+   * sidebar about the current anchoring status.
+   */
+  sync(annotations) {
+    this.bridge.call(
+      'sync',
+      annotations.map(ann => this._format(ann))
     );
-  },
-};
-
-AnnotationSync.prototype._mkCallRemotelyAndParseResults = function (
-  method,
-  callBack
-) {
-  return (function (_this) {
-    return function (annotation) {
-      // Wrap the callback function to first parse returned items
-      const wrappedCallback = function (failure, results) {
-        if (failure === null) {
-          _this._parseResults(results);
-        }
-        if (typeof callBack === 'function') {
-          callBack(failure, results);
-        }
-      };
-      // Call the remote method
-      _this.bridge.call(method, _this._format(annotation), wrappedCallback);
-    };
-  })(this);
-};
-
-// Parse returned message bodies to update cache with any changes made remotely
-AnnotationSync.prototype._parseResults = function (results) {
-  let bodies;
-  let body;
-  let i;
-  let j;
+  }
 
-  for (i = 0; i < results.length; i++) {
-    bodies = results[i];
-    bodies = [].concat(bodies);
-    for (j = 0; j < bodies.length; j++) {
-      body = bodies[j];
-      if (body !== null) {
-        this._parse(body);
-      }
+  /**
+   * Assign a non-enumerable tag to objects which cross the bridge.
+   * This tag is used to identify the objects between message.
+   *
+   * @param {string} [tag]
+   */
+  _tag(ann, tag) {
+    if (ann.$tag) {
+      return ann;
     }
+    tag = tag || window.btoa(Math.random().toString());
+    Object.defineProperty(ann, '$tag', {
+      value: tag,
+    });
+    this.cache[tag] = ann;
+    return ann;
   }
-};
 
-// Assign a non-enumerable tag to objects which cross the bridge.
-// This tag is used to identify the objects between message.
-AnnotationSync.prototype._tag = function (ann, tag) {
-  if (ann.$tag) {
-    return ann;
+  /**
+   * Copy annotation data from an RPC message into a local copy (in `this.cache`)
```

Comment:
Looks like it's actually `_tag` that does the caching. `_format` also calls `_tag`, which means it also updates the `cache` indirectly.